homeassistant/components/kef/media_player.py

```diff
@@ -137,7 +120,7 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         sources,
         speaker_type,
         loop=hass.loop,
-        unique_id=unique_id,
+        unique_id=f"kef-{host}",
```

Comment:
it is not allowed to use the host or IP for the unique IDs, see:

<https://developers.home-assistant.io/docs/entity_registry_index/#unacceptable-sources-for-a-unique-id>

Reply:
Do you have another suggestion? The examples wouldn't work, so should I use the `config_entry_id`?

Reply:
<https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort>

Yes, however, in that case, we can migrate existing entities on entry setup so it won't be a breaking change.

Reply:
I cannot find a single example that uses `async_setup_platform` and `CONF_ID` in some way.

How can I get the Config Entry ID?

Reply:
Oh right, this is a platform component with unique IDs. That is a fairly rare combination, especially without using config entries.

In that case, the unique ID of last resort doesn't apply, sorry.

Reply:
Any other ideas?

I am looking at https://developers.home-assistant.io/docs/entity_registry_index/#example-acceptable-sources-for-a-unique-id but do not see any one that would apply.

Reply:
Migrate the integration to use a config flow, as that provides a config entry of which the ID of that entry can be used.

Reply:
Unfortunately, I won't have time for that any time soon.

In the meantime, this PR is solving a real problem people are having.

Would you know a simpler solution for the time being?

Reply:
We cannot accept the current suggested solution and if the possible solutions aren't implemented right now; There is not a real other solution imho.

Reply:
What should be done with this PR at this point @basnijholt ?

Reply:
If there is no option at all to fix this without completely transferring to a config_flow, then we should close this PR. I do not see that I will have the time to do that in the near future.

Reply:
@basnijholt would it be possible to allow setting the arbitrary unique ID from the configuration.yaml? So if it is provided by a user then the mac address resolution would be skipped.

Reply:
> would it be possible to allow setting the arbitrary unique ID from the configuration.yaml? 

That is only allowed for integrations that are out of scope for ADR-0010, unfortunately, for this integration ADR-0010 does apply.

Reply:
> If there is no option at all to fix this without completely transferring to a config_flow, then we should close this PR

I'm not aware of the information the KEF integration provides, if nothing more than this, than indeed a closing of this seem inevitable at this point in time.

Reply:
An alternative is that if the device is offline, you raise `PlatformNotReady`. In that case it will be automatically retried later. In the meanwhile an `unavailable` entity will be automatically created.

Reply:
That is a great idea, thanks Paulus. I will do that!

Reply:
Thanks, I have implemented that suggestion in  #56508.